pkg/cloud/cloud.go

```diff
@@ -638,6 +642,58 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	return &Disk{CapacityGiB: size, VolumeID: volumeID, AvailabilityZone: zone, SnapshotID: snapshotID, OutpostArn: outpostArn}, nil
 }
 
+// execBatchDescribeVolumesModifications executes a batched DescribeVolumesModifications API call
+func execBatchDescribeVolumesModifications(svc EC2API, input []string) (map[string]*types.VolumeModification, error) {
+	klog.V(7).InfoS("execBatchDescribeVolumeModifications", "volumeIds", input)
+	request := &ec2.DescribeVolumesModificationsInput{
+		VolumeIds: input,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), batchDescribeTimeout)
+	defer cancel()
+
+	resp, err := describeVolumesModifications(ctx, svc, request)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]*types.VolumeModification)
+
+	for _, i := range resp {
+		volumeModification := i
+		result[*volumeModification.VolumeId] = &volumeModification
+	}
+
+	klog.V(7).InfoS("execBatchDescribeVolumeModifications: success", "result", result)
+	return result, nil
+}
+
+// batchDescribeVolumesModifications processes a DescribeVolumesModifications request by queuing the task and waiting for the result.
+func (c *cloud) batchDescribeVolumesModifications(request *ec2.DescribeVolumesModificationsInput) (*types.VolumeModification, error) {
+	var task string
+
+	if len(request.VolumeIds) == 1 && request.VolumeIds[0] != "" {
+		task = request.VolumeIds[0]
+	} else {
+		return nil, fmt.Errorf("batchDescribeVolumesModifications: invalid request, request: %v", request)
```

Comment:
Suggesting adding a unit test to cover the code path for an invalid request here.

Reply:
Great callout!  

Added new commit which adds similar unit test for all four BatchDescribe functions (as well as the `ErrInvalidRequest` err type required for this unit test to be added) so we are consistent.

I'm open to opening a separate PR for this commit because I understand changing other unit tests is out of scope.